docs/references/architecture/adr-117-multi-curve-support.md

```diff
@@ -0,0 +1,245 @@
+    vote.Signature = sig
+    return nil
+}
+
+// SignProposal signs a canonical representation of the proposal
+func (pv *CryptoProviderPV) SignProposal(chainID string, proposal *Proposal) error {
+    signer := pv.provider.GetSigner()
+
+    // code for getting proposalBytes goes here
+    // proposalBytes := ...
+
+    // The underlying signer needs these parameters so we pass them through SignerOpts
+    options := SignerOpts{
+        "chainID":  chainID,
+        "proposal": proposal,
+    }
+
+    sig, _ := signer.Sign(proposalBytes, options)
+    proposal.Signature = sig
+    return nil
+}
+
+// SignBytes signs an arbitrary array of bytes
+func (pv *CryptoProviderPV) SignBytes(bytes []byte) ([]byte, error) {
+    signer := pv.provider.GetSigner()
+    return signer.Sign(bytes, SignerOpts{})
+}
+
+```
+
+*Note:* Each provider (File, Socket, SignerClient, RetrySignerClient) will need to be modified to satisfy the `CryptoProvider` interface.
+
+
+#### Loading and storing
+
+When we refer to storing a `CryptoProvider`, we're essentially preserving two key elements: the type of crypto provider (e.g., file-based, hardware security module, remote signer, etc) and its associated configuration parameters.
+
+These configuration parameters can vary widely depending on the provider type. For instance, a file-based provider might need to store file paths, while a hardware security module provider could require device identifiers or network addresses. Remote signers might need endpoint URLs and authentication details. By storing both the type and these specific configurations, we ensure that the `CryptoProvider` can be correctly instantiated.
+
+Alternatives:
+
+* *Low impact / minimal dependencies*: The corresponding `CryptoProvider` can be directly stored on disk in a dedicated directory in an encoding format of choice (text, JSON)
+
+* *Greater impact / better security*: Use cosmos-sdk's [Keyring](https://github.com/cosmos/cosmos-sdk/blob/439f2f9d5b5884bc9df4b58d702555330549a898/crypto/keyring/keyring). to manage `CryptoProvider` along with its private keys. This specifically applies to the `FilePV` implementation, which could store its private keys through `Keyring` instead of a file in the filesystem. This approach will require decoupling the `Keyring` package from the cosmos-sdk, which could be cumbersome.
+
+
+#### Directories reorganization
+
+Implementations of crypto providers (previously `Privval` implementations) should be in their own directory:
+
+```plaintext
+cometbft/
+├── privval/
+│   ├── providers/
+│   │   ├── file_provider.go
+│   │   ├── signer_provider.go
+│   │   ├── retry_signer_provider.go
+│   │   ├── mock_provider.go
+├── types/
+│   ├── priv_validator.go
+```
+
+#### Other considerations
+
+##### Node startup
+
+Node's configuration file should include the name of the `CryptoProvider` to be loaded at startup. Also, the startup logic will need to be changed from creating a validator to loading a `CryptoProvider`
```

Comment:
Will we be loading entire crypto providers from the disk or just a name?

Reply:
The idea is to load the entire provider at startup, by loading the id we can then search for the proper file (the json that describes the provider). Once the json is loaded, it is then passed to the "Providers Factory" and its output is the provider that will be doing the signing/verification ops.

Reply:
So, the ID is a file path. Got it.